src/arrayMethodReduce.js

```diff
@@ -5,7 +5,14 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const startIndex = +(arguments.length === 1);
+    let result = startIndex ? this[0] : initialValue;
```

Comment:
It is better to avoid "result" and use more meaningful names)
```suggestion
    let accumulator = startIndex ? this[0] : initialValue;
```